libraries/freertos_plus/standard/freertos_plus_tcp/include/FreeRTOS_Sockets.h

```diff
@@ -189,11 +189,11 @@ struct freertos_sockaddr
 										( ( uint32_t ) ( ucOctet0 ) ) )
 
 	#define FreeRTOS_inet_ntoa( ulIPAddress, pucBuffer )									\
-								sprintf( ( char * ) ( pucBuffer ), "%u.%u.%u.%u",			\
-									( ( unsigned ) ( ( ulIPAddress ) & 0xffUL ) ),			\
-									( ( unsigned ) ( ( ( ulIPAddress ) >> 8 ) & 0xffUL ) ),	\
-									( ( unsigned ) ( ( ( ulIPAddress ) >> 16 ) & 0xffUL ) ),\
-									( ( unsigned ) ( ( ulIPAddress ) >> 24 ) ) )
+								sprintf( ( char * ) ( pucBuffer ), "%lu.%lu.%lu.%lu",			\
+									( ( uint32_t ) ( ( ulIPAddress ) & 0xffUL ) ),			\
+									( ( uint32_t ) ( ( ( ulIPAddress ) >> 8 ) & 0xffUL ) ),	\
+									( ( uint32_t ) ( ( ( ulIPAddress ) >> 16 ) & 0xffUL ) ),\
+									( ( uint32_t ) ( ( ulIPAddress ) >> 24 ) ) )
```

Comment:
With this change, Coverity analysis tells me that `%lu` expects `long unsigned int` whereas the argument is `unsigned int`.

Reply:
Coverity complains that "Using basic numerical type unsigned int rather than a typedef that includes size and signedness information".
Very good, but there is also a parameter check for `snprintf()`, which has a different judgement.

~~~c
/* Coverity complains about 'long' */
sprintf( pucBuffer, "%lu", ( unsigned long ) 1U );

/* Coverity complains about 'unsigned' */
sprintf( pucBuffer, "%u",  ( unsigned ) 1U );

/* Coverity complains about '%lu' */
sprintf( pucBuffer, "%lu", ( uint32_t ) 1U );

/* Compilers in some environments complain about '%u' and want '%lu' */
sprintf( pucBuffer, "%u",  ( uint32_t ) 1U );
~~~

I've been facing this problems for years already :-(
In some environments, `uint32_t` is an `unsigned long`, whereas in other ports, it is an `unsigned`.